db/global.go

```diff
@@ -0,0 +1,79 @@
+package db
+
+// NewGlobal constructs a new global object.
+func NewGlobal() *Global {
+	return &Global{}
+}
+
+// Path returns the path to the global configuration
+func (g *Global) Path() (string, error) {
+	return rootGlobal, nil
+}
+
+// Prefix returns the path to the global configuration
+func (g *Global) Prefix() string {
+	return ""
+}
+
+// Validate validates the global configuration.
+func (g *Global) Validate() error {
+	if g.MountPath == "" {
+		g.MountPath = DefaultMountPath
+	}
+
+	if g.TTL < TTLFixBase {
+		g.TTL = DefaultGlobalTTL
+	}
+
+	if g.Timeout < TimeoutFixBase {
+		g.Timeout = DefaultTimeout
+	}
+
+	return nil
+}
+
+// Hooks returns an empty hooks set.
+func (g *Global) Hooks() *Hooks {
+	return &Hooks{}
+}
+
+// Copy returns a copy of this Global.
+func (g *Global) Copy() Entity {
```

Comment:
@erikh Does the return type seem correct here?

I am trying to understand this,
```
type Global Struct {
    Debug     bool
    Timeout   time.Duration
    TTL       time.Duration
     MountPath string
}
```

Vs

```
type Entity interface {
Prefix() string
Path() (string, error)
Validate() error
Copy() Entity
Hooks() *Hooks
}
```